From 8074c987b46434282a52706e97848dda0397b22a Mon Sep 17 00:00:00 2001
From: Frédéric Moulins <frederic.moulins@alsatis.com>
Date: Fri, 28 Nov 2008 22:12:02 -0800
Subject: pppol2tp: Add missing sock_put() in pppol2tp_release()

From: Frédéric Moulins <frederic.moulins@alsatis.com>

[ Upstream commit e6358135147807351db3b7782d3e198a1bba8b62 ]

pppol2tp_sock_to_session() do sock_hold() if the session to release is
not NULL.

Signed-off-by: Frédéric Moulins <frederic.moulins@alsatis.com>
Signed-off-by: David S. Miller <davem@davemloft.net>

---
 drivers/net/pppol2tp.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/net/pppol2tp.c
+++ b/drivers/net/pppol2tp.c
@@ -1353,6 +1353,7 @@ static int pppol2tp_release(struct socke
 			kfree_skb(skb);
 			sock_put(sk);
 		}
+		sock_put(sk);
 	}
 
 	release_sock(sk);
